Devops

benjamistan opened the pull request:
Added buildspec.yml file to facilitate AWS CI/CD Pipeline.

benjamistan committed:
Update README.md

Updated README to show this as devops-only branch.

benjamistan committed:
Update README.md

benjamistan committed:
Update README.md

benjamistan committed:
feat: added buildspec.yml

benjamistan committed:
Created and refined buildspec.yaml for AWS CI/CD.

benjamistan committed:
Refined buildspec.yml

benjamistan committed:
feat: added buildspec.yml

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
refined buildspec

benjamistan committed:
updated buildspec

v-stickykeys commented on README.md:
````diff
@@ -1,3 +1,9 @@
+```diff
+- This is a DevOps ephemeral branch for configuring the CI/CD in AWS. There are no new features or code. 
````
Not sure I understand this... This pr is being merged into develop so this message isn't applicable. Anyway dont think we need different messages on different branches. And dont need diff branch for ci/cd.

v-stickykeys commented on README.md:
````diff
@@ -1,3 +1,9 @@
+```diff
+- This is a DevOps ephemeral branch for configuring the CI/CD in AWS. There are no new features or code. 
+- Nothing to see here. To use GodMode-Ganache-CLI, the Develop branch is where you want to be
````
If needed let's just make this more clear like, `Use this library from `develop` branch`

v-stickykeys commented on buildspec.yml:
```diff
@@ -0,0 +1,35 @@
+
+env:
+  parameter-store:
+    AWS_ACCOUNT_ID: account_id
```
How does this work? Is it coming from github secrets? If not that's an easy way to get this.

v-stickykeys commented on buildspec.yml:
```diff
@@ -0,0 +1,35 @@
+      - echo Writing image definitions file...
+      - printf '[{"name":"godmode-ganache-cli","imageUri":"%s"}]' $REPOSITORY_URI:$IMAGE_TAG > imagedefinitions.json
+artifacts:
+  files: imagedefinitions.json
```
Do we need an artifact? What is it used for?

v-stickykeys reviewed (changes requested):
Great overall!

Besides the code questions a few clean up things:

- Make your commit messages more descriptive so that we actually know how each commit changes behavior of the pipeline

- If the commits feel like they should have the same message do an interactive rebase and just fold them into a single commit

- Since we do have the account id in commit history right now do an interactive rebase so that the account id gets removed from git history

This is a reference that I like for rebasing https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#squash-commits-together

v-stickykeys commented on buildspec.yml:
```diff
@@ -0,0 +1,35 @@
+      - echo Build started on `date`
+      - echo Building the Docker image...
+      - docker build -t $REPOSITORY_URI:latest .
+      - docker tag $REPOSITORY_URI:latest $REPOSITORY_URI:$IMAGE_TAG
```
Can we also include branch name as a tag?

benjamistan committed:
refined buildspec
Adding Buildspec.yml for AWS CI/CD. This commit rebases multiple previous ones.

benjamistan committed:
Merge branch 'devops' of https://github.com/xGodMode/godmode-ganache-cli into devops

benjamistan commented:
Closing. This is a mess indeed. At this point it's more productive for me to try again from scratch than to unpick this chaos.
